ultravox/tools/ds_tool/ds_tool.py

```diff
@@ -131,55 +173,57 @@ class DatasetToolArgs:
     )
 
     def __post_init__(self):
-        assert (
-            not self.upload_split or self.dataset_split
-        ), "Must specify dataset_split when using upload_split"
+        assert self.dataset_subset, "dataset_subset must be specified"
+        if not self.upload_subset:
+            self.upload_subset = self.dataset_subset
+        if self.dataset_split and not self.upload_split:
+            self.upload_split = self.dataset_split
 
 
 def main(args: DatasetToolArgs):
     ds_name = args.dataset_name
     print(f'Loading dataset "{ds_name}" for task {args.task}')
-    data_dict: datasets.DatasetDict = datasets.load_dataset(
+    ds: datasets.DatasetDict = datasets.load_dataset(
         ds_name, args.dataset_subset, split=args.dataset_split
     )
-    if args.dataset_split:
-        data_dict = datasets.DatasetDict(**{args.dataset_split: data_dict})
 
-    for split, ds_split in data_dict.items():
-        print(f'Processing split "{split}"...')
-        if args.shuffle:
-            ds_split = ds_split.shuffle(seed=args.shuffle_seed)
-        if args.num_samples:
-            ds_split = ds_split.select(range(args.num_samples))
-        data_dict[split] = args.task.map_split(ds_split, args.num_workers)
+    if isinstance(ds, datasets.Dataset):
+        ds = datasets.DatasetDict({args.upload_split: ds})
+
+    if len(ds) > 1 and args.upload_split:
+        raise ValueError("Cannot upload multiple splits to a single split")
 
-    token = args.token or os.environ.get("HF_TOKEN")
     hub_args: Dict[str, Any] = {
-        "config_name": args.dataset_subset or "default",
-        "token": token,
+        "config_name": args.upload_subset,
+        "token": args.token or os.environ.get("HF_TOKEN"),
         "revision": args.upload_branch,
         "private": args.private,
-        "split": args.upload_split,
```

Comment:
I think this removes the ability to just update a single split without touching other splits in a dataset.

Reply:
The specified subset/config of the dataset is processed one split a time, and the push_to_hub call always includes the split name (original or new) in its arguments:

ds_split.push_to_hub(args.upload_name, split=upload_split, **hub_args)